packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -147,15 +149,19 @@ export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
     const currentId = id.current;
     setIsCreatingSlices(true);
     addSlices(newSlices)
-      .then(() => {
+      .then(({ slices, libraryName }) => {
         if (currentId !== id.current) return;
         setIsCreatingSlices(false);
-        // TODO: Execute modal callback.
+        onSuccess(slices, libraryName);
```

Comment:
we need to reset the modal id here as well
```suggestion
        id.current = crypto.randomUUID();
        onSuccess(slices, libraryName);
```

maybe it's a good idea to have a reset fn that puts back the common things

Reply:
I added the snippet you mentioned, but I'm not sure what you mean by adding a reset fn. Can you elaborate on that?

Reply:
we have at least a couple of places (when we're about to close the modal and when we're about to call `onSuccess`) where we have to repeat the same logic, to _reset_ the modal to its original state, preparing it to be open again with a fresh instance - i thought it was a good idea to encapsulate this logic in a function named `reset` that you can call from both places, that way, if tomorrow we need to add some logic there, we'll do it in a single place and we won't forget to add it to the other branches